terraform/staging/main.tf

```diff
@@ -133,4 +133,7 @@ module "staging_cluster" {
   provision_opensearch_cluster = false
 
   ses_email_domain = "staging.spack.io"
+
+  public_binary_mirrors       = false
+  binary_mirror_bucket_suffix = "-staging"
```

Comment:
Could this not just be based on `deployment_name`, and avoid the variable altogether?

Reply:
Since the prod bucket already exists and is called `spack-binaries`, the alternative is a conditional like this https://github.com/spack/spack-infrastructure/blob/main/terraform/modules/spack/opensearch.tf#L2. My thinking was that an explicit input var is better than assigning some implicit special meaning to the `"prod"` deployment name string.

Reply:
Ultimately these modules are really a convenience for ourselves, and we already take some special care when it comes to the prod deployment. I think it's okay to handle the "prod" deployment in a special way like you highlighted, as that still generalizes to any more deployments we make, and I'd prefer to avoid adding superfluous variables to our terraform config.

But that's just my two cents. If you see your approach as less of a technical burden then I won't block on it.

Reply:
That makes sense, I think it's fine to handle the "prod" deployment as a special case, as long as we're making an explicit decision to do that. And also, considering that I removed the other new variable based on Dan's suggestion above, it's nice to remove this one as well and avoid introducing any new vars in this PR.